ClosedXML/Excel/Ranges/XLRangeBase.cs

```diff
@@ -2067,5 +2067,38 @@ public void Select()
         {
             Worksheet.SelectedRanges.Add(AsRange());
         }
+
+        /// <summary>
+        /// Get cropped range.
+        /// </summary>
+        public IXLRangeBase Crop(IXLRangeBase crop)
+        {
+            var sheet = this.Worksheet;
+            using (var xlRange = sheet.Range(
+                Math.Max(this.RangeAddress.FirstAddress.RowNumber, crop.RangeAddress.FirstAddress.RowNumber),
+                Math.Max(this.RangeAddress.FirstAddress.ColumnNumber, crop.RangeAddress.FirstAddress.ColumnNumber),
+                Math.Min(this.RangeAddress.LastAddress.RowNumber, crop.RangeAddress.LastAddress.RowNumber),
+                Math.Min(this.RangeAddress.LastAddress.ColumnNumber, crop.RangeAddress.LastAddress.ColumnNumber)))
+            {
+                return sheet.Range(xlRange.RangeAddress);
+            }
+        }
+
+        /// <summary>
+        /// Get range relative to another range.
+        /// </summary>
+        /// <param name="baseRange">Coordinate system. Coordinates are calculated relative to this range.</param>
+        /// <param name="targetBase"></param>
+        public IXLRange Relative(IXLRangeBase baseRange, IXLRangeBase targetBase)
+        {
```

Comment:
Does `baseRange` and `targetRange` have to be the same size? If so, please add error handlers. If not, what happens if it leads to negative numbers below?

Reply:
Also, I don't quite the use of this function. It looks like something that will be required in really rare cases. Why would this need to be included in `ClosedXML` instead of leaving it to the user to calculate? Please explain.

Reply:
I often use this function in my library, so I thought that it might be necessary for someone else. If you do not think so, then I will delete it.